nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -35,21 +36,23 @@ public class WorkflowDefinitionService {
   private static final Logger logger = getLogger(WorkflowDefinitionService.class);
 
   private final Map<String, WorkflowDefinition> workflowDefinitions = synchronizedMap(new LinkedHashMap<>());
-  private List<WorkflowDefinition> workflowDefinitionValues = emptyList();
+  private List<WorkflowDefinition> workflowDefinitionValues = unmodifiableList(emptyList());
   private final WorkflowDefinitionDao workflowDefinitionDao;
   private final boolean persistWorkflowDefinitions;
   private final boolean autoInit;
   /**
    * The next time the stored definitions from database are checked for changes.
    */
   private long nextCheckOfStoredDefinitions;
+  private long storedDefinitionCheckInterval;
 
   @Inject
   public WorkflowDefinitionService(WorkflowDefinitionDao workflowDefinitionDao, Environment env) {
     this.workflowDefinitionDao = workflowDefinitionDao;
     this.persistWorkflowDefinitions = env.getRequiredProperty("nflow.definition.persist", Boolean.class);
     this.autoInit = env.getRequiredProperty("nflow.autoinit", Boolean.class);
-    this.nextCheckOfStoredDefinitions = env.getProperty("nflow.definition.load", Boolean.class, true) ? 0 : MAX_VALUE;
+    this.storedDefinitionCheckInterval = SECONDS.toMillis(env.getProperty("nflow.definition.load.interval", Integer.class, 60));
```

Comment:
It would be better to use period format here. If not, at least the config parameter name should say the unit is seconds. And the variable in code should say the unit is milliseconds.